app/actions/rmwhub.py

```diff
@@ -0,0 +1,195 @@
+                        "label": "a" if trap.get("sequence") == 0 else "b",
+                        "location": {
+                            "latitude": trap.get("latitude"),
+                            "longitude": trap.get("longitude"),
+                        },
+                    }
+                )
+
+            subject_name = subject_name + "_0"
+            latitude = update.get("traps")[0].get("latitude")
+            longitude = update.get("traps")[0].get("longitude")
+            observations.append(
+                {
+                    "name": update.get("set_id"),
+                    "source": "rmwhub_" + update.get("set_id") + "_0",
+                    "type": source_type,
+                    "subject_type": subject_subtype,
+                    "recorded_at": last_updated,
+                    "location": {"lat": latitude, "lon": longitude},
+                    "additional": {
+                        "subject_name": subject_name,
+                        "rmwHub_id": update.get("set_id"),
+                        "display_id": display_id_hash,
+                        "event_type": event_type,
+                        "devices": devices,
+                    },
+                }
+            )
+
+            if update.get("deployment_type") == "trawl":
+                subject_name = subject_name + "_1"
+                latitude = update.get("traps")[1].get("latitude")
+                longitude = update.get("traps")[1].get("longitude")
+                observations.append(
+                    {
+                        "name": update.get("set_id"),
+                        "source": "rmwhub_" + update.get("set_id") + "_1",
+                        "type": source_type,
+                        "subject_type": subject_subtype,
+                        "recorded_at": last_updated,
+                        "location": {"lat": latitude, "lon": longitude},
+                        "additional": {
+                            "subject_name": subject_name,
+                            "rmwHub_id": update.get("set_id"),
+                            "display_id": display_id_hash,
+                            "event_type": event_type,
+                            "devices": devices,
+                        },
+                    }
+                )
+
+                # TODO; Write trawl line event to ER, not required for beta
+
+        return observations
+
+    def process_deletes(self, deletes):
+        """
+        Process the deletes from the RMW Hub API.
+        """
+        pass
+
+    def upload_data():
+        pass
+
+    def delete_data():
+        pass
+
+    def sync_data():
+        pass
+
+
+class RmwHubClient:
+    HEADERS = {"accept": "application/json", "Content-Type": "application/json"}
+
+    def __init__(self, api_key: str, rmw_url: str):
+        self.api_key = api_key
+        self.rmw_url = rmw_url
+
+    def search_others(self, start_datetime_str: str) -> dict:
```

Comment:
It's probably not super critical for this particular integration, but consider making async requests for these classes in the future. You can use `httpx.AsyncClient`. It'll change the logic only a little and will perform better in the serverless environment.